apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/ApolloClient.kt

```diff
@@ -541,24 +541,15 @@ private constructor(
           sendDocument = sendDocument,
           enableAutoPersistedQueries = enableAutoPersistedQueries,
           canBeBatched = canBeBatched,
+
+          // Keep a reference to this Builder shortcuts so newBuilder() behaves as expected
+          builder = this,
       )
     }
   }
 
   fun newBuilder(): Builder {
-    return Builder()
-        .networkTransport(networkTransport)
-        .subscriptionNetworkTransport(subscriptionNetworkTransport)
-        .customScalarAdapters(customScalarAdapters)
-        .interceptors(interceptors)
-        .requestedDispatcher(requestedDispatcher)
-        .executionContext(executionContext)
-        .httpMethod(httpMethod)
-        .httpHeaders(httpHeaders)
-        .sendApqExtensions(sendApqExtensions)
-        .sendDocument(sendDocument)
-        .enableAutoPersistedQueries(enableAutoPersistedQueries)
-        .canBeBatched(canBeBatched)
+    return builder
```

Comment:
This needs to copy the builder else doing

```
val builder1 = apolloClient.newBuilder()
val builder2 = apolloClient.newBuilder()

builder1.serverUrl("https://example.com/graphql1")
// builder2.serverUrl has changed too

```

And since copying will require some boilerplate, might as well keep everything under `ApolloClient` to avoid storing properties twice?

Reply:
Ah you're right, this could be misleading.  For the sake of comparison I've tried the `copy` method in the [last commit](https://github.com/apollographql/apollo-kotlin/pull/3771/commits/a4bfcf7f461c4e6b9da1dd773352f42a3048e70e), I think the code looks a bit nicer than having this logic in `ApolloClient`, but let me know what you think.

Reply:
Works for me!